### ISSUE TITLE
V1: Seg Fault when right clicking cable to remove on analyzer-xl

### ISSUE THREAD

**Issue by Coirt** [bogaudio/BogaudioModules#56]
OS: win7
commit: a46a875

Settings changed in the analyzser context menu: amp = -120, Quality = Ultra
Crashed when trying to remove the cable ran make debug: 
```
[4.632 info src/main.cpp:161] Starting engine
[4.632 info src/main.cpp:173] Running window
[New Thread 192.0x2420]
[New Thread 192.0x1914]
[New Thread 192.0xbcc]
[Thread 192.0x1914 exited with code 0]
[Thread 192.0xbcc exited with code 0]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x000000000cc44ed5 in bogaudio::AnalyzerDisplay::binValueToHeight (
    rangeDb=140,
    value=<error reading variable: Cannot access memory at address 0x20316f64>, this=0x1e6683d0) at C:/msys64/mingw64/x86_64-w64-mingw32/include/math.h:892
892       __CRT_INLINE float powf(float _X,float _Y) { return ((float)pow((double)_X,(double)_Y)); }
(gdb) bt
#0  0x000000000cc44ed5 in bogaudio::AnalyzerDisplay::binValueToHeight (
    rangeDb=140,
    value=<error reading variable: Cannot access memory at address 0x20316f64>, this=0x1e6683d0) at C:/msys64/mingw64/x86_64-w64-mingw32/include/math.h:892
#1  bogaudio::AnalyzerDisplay::drawGraph (binsN=<optimized out>, rangeDb=140,
    rangeMaxHz=<optimized out>, rangeMinHz=<optimized out>,
    strokeWidth=<optimized out>, color=..., bins=<optimized out>, args=...,
    this=0x1e6683d0) at src/analyzer_base.cpp:473
#2  bogaudio::AnalyzerDisplay::draw (this=0x1e6683d0, args=...)
    at src/analyzer_base.cpp:240
#3  0x0000000000000000 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)
(gdb)
```

**Comment by mdemanett** [comment]
This should be fixed now.  I wasn't able to replicate it, though clearly it can happen...if you have a minute to test and let me know if it seems to be working, that would be a great help.

**Comment by Coirt** [comment]
I have not been able to reproduce either. Possibly might have been indirectly fixed with a newer rack commit. If it happens again I will reopen.

**Comment by Coirt** [comment]
Hmm I managed to break it again.

I was changing the window modes between kaiser, hamming and none. Same Seg Fault occurred on make run of rack. 

some more info 
I noticed if you have two cables connected, the colours for each channels frequency response do not update when changing modes in the context menu, they stay the same colour as the bottom most connected input, it is only when you disconnect a cable that the window updates the colours.

Iv'e been trying to reproduce with one cable connected perhaps it is caused by the union between the frequency response colours on the graph. This happens when changing quality and window modes not sure about other settings.
https://imgur.com/izTauu2
